apps/server/src/modules/group/controller/api-test/group.api.spec.ts

```diff
@@ -157,4 +158,121 @@ describe('Group (API)', () => {
 			});
 		});
 	});
+
+	describe('getGroup', () => {
+		describe('when user with the required permission requests a group', () => {
+			describe('when group exists', () => {
+				const setup = async () => {
+					const school: SchoolEntity = schoolFactory.buildWithId();
+					const { teacherAccount, teacherUser } = UserAndAccountTestFactory.buildTeacher({ school });
+
+					const group: GroupEntity = groupEntityFactory.buildWithId({
+						users: [
+							{
+								user: teacherUser,
+								role: teacherUser.roles[0],
+							},
+						],
+					});
+
+					await em.persistAndFlush([teacherAccount, teacherUser, group]);
+					em.clear();
+
+					const loggedInClient = await testApiClient.login(teacherAccount);
+
+					return {
+						loggedInClient,
+						group,
+						teacherUser,
+					};
+				};
+
+				it('should return the group', async () => {
+					const { loggedInClient, group, teacherUser } = await setup();
+
+					const response = await loggedInClient.get(`${group.id}`);
+
+					expect(response.status).toEqual(HttpStatus.OK);
+					expect(response.body).toEqual({
+						id: group.id,
+						name: group.name,
+						type: group.type,
+						users: [
+							{
+								id: teacherUser.id,
+								firstName: teacherUser.firstName,
+								lastName: teacherUser.lastName,
+								role: teacherUser.roles[0].name,
+							},
+						],
+						externalSource: {
+							externalId: group.externalSource?.externalId,
+							systemId: group.externalSource?.system.id,
+						},
+					});
+				});
+			});
+
+			describe('when group does not exist', () => {
+				const setup = async () => {
+					const { teacherAccount, teacherUser } = UserAndAccountTestFactory.buildTeacher();
+
+					await em.persistAndFlush([teacherAccount, teacherUser]);
+					em.clear();
+
+					const loggedInClient = await testApiClient.login(teacherAccount);
+
+					return {
+						loggedInClient,
+					};
+				};
+
+				it('should return not found', async () => {
+					const { loggedInClient } = await setup();
+
+					const response = await loggedInClient.get(`${new ObjectId().toHexString()}`);
+
+					expect(response.status).toEqual(HttpStatus.NOT_FOUND);
+					expect(response.body).toEqual({
+						code: HttpStatus.NOT_FOUND,
+						message: 'Not Found',
+						title: 'Not Found',
+						type: 'NOT_FOUND',
+					});
+				});
+			});
+		});
+
+		describe('when user without the required permission requests a group', () => {
```

Comment:
I think we are supposed to test for unauthorized and not for forbidden. Permission checks (authorization) are tested in the uc and authentication is tested in the api test.